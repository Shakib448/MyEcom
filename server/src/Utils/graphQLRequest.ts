import app from "./server";
import supertest from "supertest";

const graphQLRequest = (query: string, token?: string) => {
  return supertest(app)
    .post("/graphql")
    .send({
      query,
    })
    .set("Authorization", `Bearer ${token}`);
};

export default graphQLRequest;
