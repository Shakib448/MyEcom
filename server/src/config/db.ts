import mongoose from "mongoose";
import "colors";

const dbURL = "mongodb://localhost:27017/myEcom";

export async function connectDB(): Promise<any> {
  await mongoose.connect(dbURL);
  console.log("Database connected successfully".cyan.italic);
}

export async function stopDB(): Promise<void> {
  await mongoose.connection.close();
}
