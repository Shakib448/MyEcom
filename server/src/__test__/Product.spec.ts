import { connectDB, stopDB } from "../config/db";
import Product from "../models/Product.model";
import graphQLRequest from "../Utils/graphQLRequest";
import products from "../data/products.data";

describe("Product Testing", () => {
  beforeAll(async () => {
    await connectDB();
  });
  afterAll(async () => {
    await stopDB();
  });

  it("fetch all products with success", async () => {
    const res = await graphQLRequest(`query {
        getAllProducts {
        id
          name
          image
          description
          brand
          category
          price
          countInStock
          rating
          numReviews
        }
      }`);

    res.body.data.getAllProducts.map((item: any) => {
      expect(typeof item.id).toBe("string");
      expect(typeof item.name).toBe("string");
      expect(typeof item.image).toBe("string");
      expect(typeof item.description).toBe("string");
      expect(typeof item.brand).toBe("string");
      expect(typeof item.category).toBe("string");
      expect(typeof item.price).toBe("number");
      expect(typeof item.countInStock).toBe("number");
      expect(typeof item.rating).toBe("number");
      expect(typeof item.numReviews).toBe("number");
    });
    expect(Array.isArray(res.body.data.getAllProducts)).toBe(true);
    expect(res.statusCode).toBe(200);
  });
});
