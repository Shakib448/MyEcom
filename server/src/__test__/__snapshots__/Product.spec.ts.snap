// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Testing Product will be created with success message 1`] = `
Object {
  "brand": "sample brand",
  "category": "sample category",
  "countInStock": 10,
  "description": "sample description",
  "image": "https://i.ibb.co/q9QZnkL/airpods.jpg",
  "name": "simple",
  "numReviews": 0,
  "price": 89.99,
  "rating": 0,
  "reviews": Array [],
}
`;
